experimental/overhead_matching/swag/data/vigor_dataset.py

```diff
@@ -404,8 +407,12 @@ def __getitem__(self, idx_or_pair: int | SamplePair):
 
         pano_metadata = self._panorama_metadata.loc[pano_idx]
         sat_metadata = self._satellite_metadata.loc[sat_idx]
-        pano, pano_original_shape = load_image(pano_metadata.path, self._panorama_size)
-        sat, sat_original_shape = load_image(sat_metadata.path, self._satellite_patch_size)
+        if self._config.should_load_images:
+            pano, _ = load_image(pano_metadata.path, self._panorama_size)
+            sat, _ = load_image(sat_metadata.path, self._satellite_patch_size)
+        else:
+            pano = torch.zeros((0, *self._panorama_size))
```

Comment:
maybe fill with nan/make sure we don't accidentally train on empty images if we aren't expecting to?